presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftBufferResult.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.facebook.drift.annotations.ThriftStruct;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public class ThriftBufferResult
+{
+    private final String taskInstanceId;
+    private final long token;
+    private final long nextToken;
+    private final boolean bufferComplete;
+    private final List<ThriftSerializedPage> thriftSerializedPages;
+
+    public static ThriftBufferResult createThriftBufferResult(BufferResult bufferResult)
```

Comment:
nit: what about `from` or `fromBufferResult`?  This seems to be the convention for factory method name in Presto, for example, see `RowType::from`, `FieldId::from`, etc :)